pkgs/tools/admin/pgadmin/default.nix

```
@@ -84,6 +84,18 @@ let
         sha256 = "sha256-FZcuUBffBXXD1sCQuhaLbbkCWeYgrI1+qBOjlrrVtss=";
       };
     });
+    # pgadmin 6.14 is incompatible with the major flask-security-too update to 5.0.x
+    flask-security-too = prev.flask-security-too.overridePythonAttrs (oldAttrs: rec {
+      version = "4.1.5";
+      src = oldAttrs.src.override {
+        inherit version;
+        sha256 = "sha256-98jKcHDv/+mls7QVWeGvGcmoYOGCspxM7w5/2RjJxoM=";
+      };
+      propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [
+        pythonPackages.flask_mail
+        pythonPackages.pyqrcode
```

I think you want to use prev/final here

---

I'm unclear why we shouldn't use `oldAttrs` here. A little bit above I used `prev` in `flask-security-too = prev.flask-security-too.overridePythonAttrs (oldAttrs: rec {` 
Could you elaborate?

---

hi @SuperSandro2000 pgadmin4 currently fails due to this issue and a build failure of `azure-mgmt-subscription`  (which gets fixed by #193926). I'd love to see this PR get merged quickly, so `pgadmin` builds again

---

```suggestion
      propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [
        final.flask_mail
        final.pyqrcode
```
I mean like this. Otherwise we might get infinite recursion.

---

Thanks a lot for the example and explanation! I applied your recommendation and rebased